upstream.go

```diff
@@ -981,6 +993,24 @@ func (uc *upstreamConn) handleMessage(msg *irc.Message) error {
 				})
 			}
 		}
+	case irc.ERR_MONLISTFULL:
+		var targets string
+		if err := parseMessageParams(msg, nil, nil, &targets); err != nil {
+			return err
+		}
+		for _, target := range strings.Split(targets, ",") {
+			uc.network.monitors.Delete(irc.ParsePrefix(target).Name)
+		}
+		fallthrough
+	case irc.RPL_MONONLINE, irc.RPL_MONOFFLINE, irc.RPL_MONLIST, irc.RPL_ENDOFMONLIST:
+		// We only support single-upstream mode for MONITOR for now
+		uc.forEachDownstream(func(dc *downstreamConn) {
+			if dc.upstream() == nil {
+				return
+			}
+
+			dc.SendMessage(msg)
```

Comment:
Shouldn't this be sent only to downstreams that are monitoring this specific target?

Reply:
I wouldn't loose sleep against broadcasting MONITOR numerics even if downstreams didn't ask for it. But we'll probably need to add per-downstream MONITOR target tracking anyways.